Code/Framework/AzToolsFramework/Tests/EditorViewportOnlyDrawHelpersForSelectedEntityTests.cpp

```diff
@@ -0,0 +1,186 @@
+            AzFramework::DebugDisplayRequestBus::BusPtr debugDisplayBus;
+            AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, TestViewportId);
+            m_debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
+
+            m_cameraState = AzFramework::CreateDefaultCamera(AZ::Transform::CreateIdentity(), AzFramework::ScreenSize(1024, 768));
+
+            m_entityVisibleEntityDataCacheMock = AZStd::make_unique<::testing::NiceMock<MockEditorVisibleEntityDataCacheInterface>>();
+            m_editorHelpers = AZStd::make_unique<AzToolsFramework::EditorHelpers>(m_entityVisibleEntityDataCacheMock.get());
+            m_viewportSettings = AZStd::make_unique<ViewportSettingsTestImpl>();
+
+            m_viewportSettings->Connect(TestViewportId);
+            m_viewportSettings->m_helpersVisible = true;
+            m_viewportSettings->m_iconsVisible = true;
+
+            using ::testing::_;
+            using ::testing::Return;
+
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, GetVisibleEntityId(_)).WillByDefault(Return(AZ::EntityId(m_entityId)));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, VisibleEntityDataCount()).WillByDefault(Return(1));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntityIconHidden(_)).WillByDefault(Return(false));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntityVisible(_)).WillByDefault(Return(true));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntitySelected(_))
+                .WillByDefault(
+                    [this](size_t)
+                    {
+                        AzToolsFramework::EntityIdList entityIds;
+                        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+                            entityIds, &AzToolsFramework::ToolsApplicationRequests::GetSelectedEntities);
+
+                        if (AZStd::ranges::find(entityIds, m_entityId) != entityIds.end())
+                        {
+                            return true;
+                        }
+                        return false;
+                    });
+        }
+
+        void TearDownEditorFixtureImpl() override
+        {
+            m_viewportSettings->Disconnect();
+            m_viewportSettings.reset();
+            m_editorHelpers.reset();
+            m_entityVisibleEntityDataCacheMock.reset();
+            AzFramework::EntityDebugDisplayEventBus::Handler::BusDisconnect();
+            AllocatorsTestFixture::TearDown();
+        }
+
+        // EntityDebugDisplayEventBus overrides ...
+        // This function is called in DisplayComponents which is responsible for drawing the helpers, if this is called it means that a
+        // helper has been drawn
+        void DisplayEntityViewport(const AzFramework::ViewportInfo&, AzFramework::DebugDisplayRequests&) override
+        {
+            m_displayEntityViewportEvent = true;
```

Comment:
update: It might be helpful to be a little more granular here and know _which_ entity was drawn (e.g. you have two entities in the camera frustum that you know of, and only one selected. We want to check we get the draw notification for that entity and not the other one when we have 'only draw selected' checked, and it's hard to be sure of that here

Reply:
I believe there's a way to find the Id of the particular notification or have a separate instance you connect to per entity to check, but I do think this would be valuable to make the test more precise